sim-lib/src/lib.rs

```diff
@@ -175,20 +194,22 @@ impl Simulation {
         // We always want to wait ofr all threads to exit, so we wait for all of them to exit and track any errors
         // that surface. It's okay if there are multiple and one is overwritten, we just want to know whether we
         // exited with an error or not.
-        let mut results = vec![];
+        let mut success = true;
         while let Some(res) = set.join_next().await {
-            results.push(res);
+            if let Err(e) = res {
+                log::error!("Task exited with error: {e}");
+                success = false;
+            }
         }
-        (!(results.iter().any(|x| matches!(x, Err(..)))))
-            .then_some(())
-            .ok_or_else(|| anyhow!("One of our tasks failed"))
+        success.then_some(()).ok_or(SimulationError::TaskError)
     }
 }
 
 // consume_events processes events that are crated for a lightning node that we can execute actions on. If it exits,
 // it will use the trigger provided to trigger shutdown in other threads. If an error occurs elsewhere, we expect the
 // senders corresponding to our receiver to be dropped, which will cause the receiver to error out and exit.
 async fn consume_events(
+    node_id: PublicKey,
```

Comment:
As for producer, let's start/stop logging for consumer. Future debugging selves will thank us.